src/grt/src/fastroute/src/utility.cpp

```diff
@@ -1765,7 +1765,7 @@ void FastRouteCore::copyRS(void)
 
   if (!sttrees_bk_.empty()) {
     for (netID = 0; netID < netCount(); netID++) {
-      if (nets_[netID]->isRouted() || nets_[netID]->isDeleted()) {
+      if (nets_[netID] == nullptr || nets_[netID]->isRouted()) {
```

Comment:
This is repeated many times so you might make a ```FastRouteCore::isRouted(int netID)``` method to encapsulate this pattern.

Reply:
I created a method `FastRouteCore::skipNet(int netID)`. Since this check isn't only checking if the net is routed I belive skipNet was a more appropriate name, if you hink other wise I can change it to `FastRouteCore::isRouted(int netID)`